source/MaterialXGenMdl/Nodes/SourceCodeNodeMdl.cpp

```diff
@@ -0,0 +1,192 @@
+            StringSet variableNames;
+            StringVec code;
+            while (i != string::npos)
+            {
+                code.push_back(_functionSource.substr(pos, i - pos));
+                size_t j = _functionSource.find_first_of(postfix, i + 2);
+                if (j == string::npos)
+                {
+                    throw ExceptionShaderGenError("Malformed inline expression in implementation for node " + node.getName());
+                }
+
+                const string variable = _functionSource.substr(i + 2, j - i - 2);
+                const ShaderInput* input = node.getInput(variable);
+                if (!input)
+                {
+                    throw ExceptionShaderGenError("Could not find an input named '" + variable +
+                        "' on node '" + node.getName() + "'");
+                }
+
+                if (input->getConnection())
+                {
+                    code.push_back(shadergen.getUpstreamResult(input, context));
+                }
+                else
+                {
+                    string variableName = node.getName() + "_" + input->getName() + "_tmp";
+                    if (!variableNames.count(variableName))
+                    {
+                        ShaderPort v(nullptr, input->getType(), variableName, input->getValue());
+                        shadergen.emitLineBegin(stage);
+                        const Syntax& syntax = shadergen.getSyntax();
+                        const string valueStr = (v.getValue() ? syntax.getValue(v.getType(), *v.getValue()) : syntax.getDefaultValue(v.getType()));
+                        const string& qualifier = syntax.getConstantQualifier();
+                        string str = qualifier.empty() ? EMPTY_STRING : qualifier + " ";
+                        str += syntax.getTypeName(v.getType()) + " " + v.getVariable();
+                        str += valueStr.empty() ? EMPTY_STRING : " = " + valueStr;
+                        shadergen.emitString(str, stage);
+                        shadergen.emitLineEnd(stage);
+                        variableNames.insert(variableName);
+                    }
+                    code.push_back(variableName);
+                }
+
+                pos = j + 2;
+                i = _functionSource.find_first_of(prefix, pos);
+            }
+            code.push_back(_functionSource.substr(pos));
+
+            shadergen.emitLineBegin(stage);
+            shadergen.emitOutput(node.getOutput(), true, false, context, stage);
+            shadergen.emitString(" = ", stage);
+            for (const string& c : code)
+            {
+                shadergen.emitString(c, stage);
+            }
+            shadergen.emitLineEnd(stage);
+        }
+        else
+        {
+            // An ordinary source code function call
+
+            shadergen.emitLineBegin(stage);
+
+            // Declare the output variables. Only output 1 for now 
```

Comment:
Redid the function call to spit out the output = function(). instead of calling function() with the output as last argument.
I think it was mentioned that we'd need to introduce struct outputs if we have a multioutput on function(), which is not handled here.

Reply:
Yep, multi-output needs to be done using structs. I have it on my todo-list to implement this.